x-pack/solutions/security/plugins/security_solution/public/common/hooks/use_navigate_to_alerts_page_with_filters.ts

```diff
@@ -16,14 +16,18 @@ import { URL_PARAM_KEY } from './use_url_state';
 export const useNavigateToAlertsPageWithFilters = () => {
   const { navigateTo } = useNavigation();
 
-  return (filterItems: FilterControlConfig | FilterControlConfig[], openInNewTab = false) => {
+  return (
+    filterItems: FilterControlConfig | FilterControlConfig[],
+    openInNewTab = false,
+    timerange?: string
+  ) => {
     const urlFilterParams = encode(
       formatPageFilterSearchParam(Array.isArray(filterItems) ? filterItems : [filterItems])
     );
-
+    const timerangePath = timerange ? `&timerange=${timerange}` : '';
     navigateTo({
       deepLinkId: SecurityPageName.alerts,
-      path: `?${URL_PARAM_KEY.pageFilter}=${urlFilterParams}`,
+      path: `?${URL_PARAM_KEY.pageFilter}=${urlFilterParams}${timerangePath}`,
```

Comment:
Im wondering if we can't use the existing `urlFilterParams` and add timerange to them directly.
The problem the might occur with the current fix is that what happens if both `urlFilterParams` and `timerangePath` include a timerange filter?

the table file seems to include the needed functions to create this query. and the encoding is happening here on this hook file. so if we can insert the timerange filter from the table it would be ideal.

This is only a code optimization, so im approving, i might also be missing the reason why this suggestion wont work so ill be glad if you could explain if you already tried it.

Reply:
hmmmm so before this actually I Tried adding timerange on the Alerts page filter to no avail
but your question is a good one, I'm not 100% sure if its not doable on adding timerange directly on the pageFilter, Im going to ask Security Team about this